src/components/AddComment.jsx

```diff
@@ -0,0 +1,65 @@
+import { useState } from "react";
+import { postComment } from "../utils/api";
+import { useParams } from "react-router-dom";
+import CommentList from "./CommentList";
+
+const AddComment = ({ setComments }) => {
+  const [author, setAuthor] = useState("");
+  const [body, setBody] = useState("");
+  const [check, setCheck] = useState(false);
+  const { article_id } = useParams();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if(body!=="" && author!=="") {
+        const optimisticComment = {
+          author: author,
+          body: body,
+          votes: 0,
+        };
+        const newComment = {
+          author: author,
+          body: body,
+        };
+        setComments((comments) => {
+          return [optimisticComment, ...comments];
+        });
+        setCheck(false);
+        postComment(article_id, newComment).then();
+    } else {
+        setCheck(true);
+    }
+  };
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <h2>Submit a Comment</h2>
+        <label htmlFor="comment-author">Author:</label>
+        <input
+          type="text"
+          id="comment-author"
+          value={author}
+          onChange={(e) => {
+            setAuthor(e.target.value);
+          }}
+        />
+        <br />
+        <label htmlFor="comment-body">Body:</label>
+        <input
```

Comment:
This may be better as a textarea rarther than input just to allow a multi line/longer comment :)